app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt

```diff
@@ -214,6 +215,41 @@ class FireDatabase : Database() {
         return updateUserData(dataUpdated)
     }
 
+    override fun addUserToFriendsList(userId: String): CompletableFuture<Boolean> {
+        val future = CompletableFuture<Boolean>()
+        isUserStoredInDatabase(userId).thenAccept {
+            if(!it){
+                future.completeExceptionally(java.lang.Error("The user with $userId is not present on the database."))
+            }else{
+                val newFriend = HashMap<String,Any?>()
+                newFriend.put(userId, null)
+                accessUserAccountFile(userId).child(friendsListFile).updateChildren(newFriend)
+                    .addOnSuccessListener { future.complete(true) }
+                    .addOnFailureListener { future.completeExceptionally(it) }
+            }
+        }
+        return future
+    }
+
+    override fun removeUserToFriendsList(userId: String): CompletableFuture<Boolean> {
+        val future = CompletableFuture<Boolean>()
+
+        accessUserAccountFile(userId).child(friendsListFile).get()
+            .addOnSuccessListener {previousFriendsList->
+                if(!previousFriendsList.children.contains(userId as DataSnapshot)){
```

Comment:
Removing a friend that isn't a friend shouldn't be an error. It technically succeeded as the resulting friend list doesn't contain the target user as expected.

Reply:
We need to control this aspect in the new implementation.

Reply:
You have right. Thank you